src/WarcraftLegacies.Source/Quests/Frostwolf/QuestHighmountain.cs

```diff
@@ -0,0 +1,54 @@
+﻿using MacroTools.QuestSystem;
+using static War3Api.Common;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.LegendSystem;
+using MacroTools.Extensions;
+using System.Collections.Generic;
+using WCSharp.Shared.Data;
+using MacroTools.FactionSystem;
+
+namespace WarcraftLegacies.Source.Quests.Frostwolf
+{
+  /// <summary>
+  /// Bring any hero to the Echo Isles to unlock the base.
+  /// </summary>
+  public sealed class QuestHighmountain : QuestData
+  {
+    private readonly LegendaryHero _cairne;
+    private readonly List<unit> _rescueUnits = new();
+
+    /// <inheritdoc />
+    public QuestHighmountain(LegendaryHero cairne, Rectangle rescueRect) : base("A Feast for Our Kin",
+      "Scouts report sighting of the Highmountain totem, thought lost long ago when the Broken Isles were shattered. As a gesture of renewed welcome, Cairne might offer them an invitation to a feast in Thunderbluff.",
+      @"ReplaceableTextures/CommandButtons/BTNPigHead.blp")
+    {
+      _cairne = cairne;
+      AddObjective(new ObjectiveLegendInRect(cairne, rescueRect, "Highmountain, north of Stormheim"));
+      ResearchId = Constants.UPGRADE_R0A9_QUEST_COMPLETED_INVITATION_TO_A_FEAST;
+      _rescueUnits = rescueRect.PrepareUnitsForRescue(RescuePreparationMode.Invulnerable);
+    }
+
+    /// <inheritdoc />
+    protected override string RewardFlavour =>
+      "Cairne is welcomed in Highmountain like a lost-long friend. Eager to explore the world and fight alongside their long-lost brethren, the Highmountain send their best hunters to support the Horde.";
```

Comment:
Now that `OnComplete` gives you the base, this should probably make some flavour reference to that.
```suggestion
      "Cairne is welcomed in Highmountain like a lost-long friend. Eager to explore the world and fight alongside their long-lost brethren, the Highmountain send their best hunters to support the Horde, and offer their home as a traveler's respite.";
```